common/src/main/java/com/microsoft/identity/common/adal/internal/tokensharing/TokenShareUtility.java

```diff
@@ -22,7 +22,7 @@
 // THE SOFTWARE.
 package com.microsoft.identity.common.adal.internal.tokensharing;
 
-import android.util.Pair;
+import com.microsoft.identity.common.java.util.ported.KeyValuePair;
```

Comment:
At this point... you've had to change every usage because the signature changed.  Why not Map.Entry?

Reply:
Replacing List<Pair<,>> with a Map<,> is a breaking change - which would require verification of the scenarios below

1. calls to eSTS (haven't got back answers, only know that they reject requests with duplicated brkr=1)
2. calls to ADFS (they support multiple query parameter values of the same key, but are not using it in any scenarios)
3. calls to on-prem identity solutions customers might be using (as pointed out by @iambmelt in the thread to ADFS, eSTS)

Given the size of coverage, I prefer not to deal with with the moment.

Reply:
I didn't say Map... I said Map.Entry.  List<KeyValyePair<A, B>> is equivalent to Iterable<Map.Entry<A, B>> as far as I can tell, but doesn't require a new class and doesn't provide list-semantics.  Why not?  Do we need List APIs?

Reply:
Not sure if I'm following.... Map.Entry is an interface, are you suggesting that I override a pair class from that?

Reply:
There are current implementations that ship with java.  Pretty sure I put AbstractMap.SimpleEntry into Common for this purpose where there wasn't already a usage of Pair and I didn't want to create a new one.

Reply:
https://docs.oracle.com/javase/7/docs/api/java/util/AbstractMap.SimpleEntry.html

Reply:
And if you change every usage of this thing to refer to the interface rather than the concrete class, just in case we want to change it in the future... you can ship it.  It might be worth having a utility method to make these, if we're calling new a lot, your call.